R/France.R

```diff
@@ -0,0 +1,142 @@
+#' @details Inherits from `DataClass`
+#' @examples
+#' \dontrun{
+#' region <- France$new(verbose = TRUE, steps = TRUE)
+#' region$download()
+#' region$clean()
+#' region$process()
+#' region$return()
+#' }
+France <- R6::R6Class("France",
+  inherit = DataClass,
+  public = list(
+
+    # Core Attributes (amend each paramater for country specific infomation)
+    #' @field level_1_region the level 1 region name.
+    level_1_region = "region",
+    #' @field level_2_region the level 1 region name.
+    level_2_region = "departement",
+    #' @field data_url link to raw data for cases
+    data_url = "https://www.data.gouv.fr/fr/datasets/r/406c6a23-e283-4300-9484-54e78c8ae675", # nolint
+    #' @field source_data_cols existing columns within the raw data
+    source_data_cols = c("cases_new", "deaths_new",
+                         "hosp_new", "tested_new"),
+    #' @field hosp_url link to raw data for hospitalisations
+    hosp_url = "https://www.data.gouv.fr/fr/datasets/r/6fadff46-9efd-4c53-942a-54aca783c30c", # nolint
+
+
+    #' @description France-specific function for downloading raw data.
+    download = function() {
+      if (self$level == 1) {
+        self$data_url <- "https://www.data.gouv.fr/fr/datasets/r/001aca18-df6a-45c8-89e6-f82d689e6c01" # nolint
+      }
+      if (self$verbose) {
+        message("Downloading data")
+        self$data$raw <- suppressWarnings(
+          csv_reader(self$data_url)
+        )
+        if (self$level == 2) {
+          self$data$raw_hosp <- suppressWarnings(
+            csv_reader(self$hosp_url)
+          )
+        }
+      } else {
+        self$data$raw <- suppressMessages(
+          suppressWarnings(
+            csv_reader(self$data_url)
+          )
+        )
+        if (self$level == 2) {
+          self$data$raw_hosp <- suppressMessages(
+            csv_reader(self$hosp_url)
+          )
+        }
+      }
+    },
+
+    #' @description *France* specific provincial/territorial level data
+    #' cleaning
+    #' @param ... pass additional arguments
+    #'
+    #' @importFrom dplyr filter select mutate rename
+    #' @importFrom tidyr replace_na
+    #' @importFrom lubridate dmy
+    clean = function(...) {
+      # function to clean the data (MUST BE CALLED clean)
+      # modify the data variable 'region' in place and add using 'self'
+      # e.g. self$data$clean <- something
+      # No return statement is required
+      # have a statement like this to indicate information to user if requested
+      if (self$verbose) {
+        message("Cleaning data")
+      }
+
+      if (self$level == "1") {
+        self$clean_level_1()
+      } else if (self$level == "2") {
+        self$clean_level_2()
+      }
+    },
+
+    #' @description France Specific Region Level Data Cleaning
+    #'
+    #' @importFrom dplyr filter mutate left_join rename select
+   clean_level_1 = function() {
+      self$data$clean <- self$data$raw %>%
+        dplyr::filter(cl_age90 == 0) %>%
```

Comment:
We've switched to not using the `package:function` framework in the dev version as we found that we were missing dependencies. Can you drop these.

Reply:
I think these have now been cleaned.